src/python/pants/engine/build_files.py

```diff
@@ -93,6 +93,12 @@ def _raise_did_you_mean(address_family: AddressFamily, name: str, source=None) -
   )
 
 
+@rule
+async def find_build_files(addresses: Addresses) -> BuildFileAddresses:
```

Comment:
Do we know which codepath this will still be used in? Will probably want to confirm that we don't see a performance impact.

Reply:
It's used solely for `LegacyAddressMapper`, which is used by `filter` I think?